include/ccf/service/tables/executor_code_id.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/ds/quote_info.h"
+#include "ccf/service/code_digest.h"
+#include "ccf/service/map.h"
+
+namespace ccf
+{
+  struct ExecutorCodeInfo
+  {
+    CodeStatus status;
+    QuoteFormat platform;
+    // FIXLater: Add more metadata that includes allowed URIS for each executor
```

Comment:
What are the allowed URIs for each executor? I'm not sure why we want to have them, considering we expect the executors to be clients and to connect to us.

Meta: we try to not to have TODO/FIXME etc comments in the CCF project, and instead we prefer to file tickets, and TODO is one template for them. We didn't always do it that way, but we observed that TODO comments were not always getting addressed or prioritized in a timely manner.

Reply:
@achamayou There's 2 kinds of metadata I think we'll want to set here - what URIs is this CodeID allowed to handle, and what tables is this CodeID allowed to access. When the CodeID completely describes the executor's behaviour (ie - it's compiled and has no dynamically loaded code), these are unnecessary but maybe useful (you don't _need_ to do a full code audit, if you trust this new executor is limited to only accessing its "own" tables). It's possible we'll also want to support executors that dynamically load their code in the same style as `js_generic`, in which case the CodeID tells us nothing about their behaviour. I think that limiting the tables and URIs the executor can own is then a useful restriction.

Simpler answer: If everything becomes a remote-executor app, then this is a nice place to implement #4096.

Reply:
The `executor` here refers to the business logic containers. The allowed URIs will let BL containers register themselves for specific URIs.

> we try to not to have TODO/FIXME etc comments in the CCF project, and instead we prefer to file tickets,

Ah got it! Though the TODO here was more of a note to the reviewers since this is an evolving change for the demo. Removed it though, will probably put it into a ticket.